pkg/controller/node/nodecontroller.go

```diff
@@ -573,6 +668,42 @@ func (nc *NodeController) monitorNodeStatus() error {
 	return nil
 }
 
+// evictIntolerablePods checks for given node if there's any pod that
```

Comment:
Can you write a comment describing when this gets invoked? I assume it will definitely be invoked any time node controller adds a taint to a node? How about when operator manually adds a taint to a node through API server? Will something reasonable happen if a taint is *removed* from a node? Is it possible to update a pod to add/remove a toleration, and if so, will the right thing happen in those scenarios?